src/EnergyPlus/SQLiteProcedures.hh

```diff
@@ -56,10 +56,10 @@ public:
 	// This allows for testing of private methods in SQLite
 	friend class SQLiteFixture;
 
-	void addScheduleData( int const number, std::string const name, std::string const type, double const minValue, double const maxValue );
+	void addScheduleData( int const number, std::string const & name, std::string const & type, double const minValue, double const maxValue );
```

Comment:
@mbadams5 Thanks for bringing this issue up. That certainly sounds like surprising behavior. If the strings are in scope the C++ lifetime rules prevent them from being deallocated unless some other code is doing something very naughty. In a multithreaded context another thread could certainly clear or modify the value of those strings but that shouldn't be happening here in unit tests. If the SQL writes are done in a secondary thread then, yes, that could gum things up but then I'd expect we would need to pass all the arguments by value, but that wasn't happening. If you know of a case that I can reproduce this with I'm happy to investigate. I did run the unit tests on this branch on Linux and it passed. I also ran my suite of test cases but I don't normally try to verify the sql output: the one I dumped and compared seemed fine. But if threading/timing is involved then that isn't enough to be sure it is safe.

Passing const strings around by value is a very bad example for our new-to-C++ developers so, yes, I was cleaning these up in a few places. Doing that as a work-around to mask some other code problem is not advisable either, but if we find an odd situation where we must then it should be comment-flagged. This doesn't mean we can't do cool, modern value semantics (Sean Parent) stuff if we really find a need for it but that would also need commenting to avoid confusing developers.

If we can find clarifying information about SQL writes (always threaded, can be done non-threaded, ...?) then we'll have enough information on how to make the SQL interfaces safe and documented.